### PR TITLE
[AUTHZ] Support create table command for Paimon

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

Apache Paimon is an incubating Apache project of data lake platform for high-speed data ingestion, changelog tracking and efficient real-time analytics.

- Initial support for Paimon tables in Authz plugin
  - Create Table Command: https://paimon.apache.org/docs/master/engines/spark3/#create-table
- Paimon `0.5.0-incubating` supports Spark 3.1/3.2/3.3/3.4.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/PaimonCatalogRangerSparkExtensionSuite.scala [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import org.apache.kyuubi.plugin.spark.authz.util.AuthZUtils._
+import org.apache.kyuubi.tags.PaimonTest
+import org.apache.kyuubi.util.AssertionUtils._
+
+/**
+ * Tests for RangerSparkExtensionSuite on Paimon
+ */
+@PaimonTest
+class PaimonCatalogRangerSparkExtensionSuite extends RangerSparkExtensionSuite {
+  override protected val catalogImpl: String = "hive"
```
Comment (SteNicholas) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/PaimonCatalogRangerSparkExtensionSuite.scala:33]:
```suggestion
  override protected val catalogImpl: String = "paimon"
```

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/PaimonCatalogRangerSparkExtensionSuite.scala [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  override def beforeAll(): Unit = {
+    if (isSupportedVersion) {
+      spark.conf.set(s"spark.sql.catalog.$catalogV2", "org.apache.paimon.spark.SparkCatalog")
+      spark.conf.set(
+        s"spark.sql.catalog.$catalogV2.warehouse",
+        Utils.createTempDir("iceberg-hadoop").toString)
+      super.beforeAll()
+    }
+
+    doAs(admin, sql(s"CREATE DATABASE IF NOT EXISTS $catalogV2.$namespace1"))
```
Comment (SteNicholas) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/PaimonCatalogRangerSparkExtensionSuite.scala:54]: Adds drop database in `afterAll`.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/PaimonCatalogRangerSparkExtensionSuite.scala (hunk 1 above)
Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/PaimonCatalogRangerSparkExtensionSuite.scala:33]: This is not directly config for V2 catalog implementation of Paimon (see relative configs preparation in beforeAll). Similar legacy config could be found in suites for Iceberg and Hudi.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/PaimonCatalogRangerSparkExtensionSuite.scala (hunk 2 above)
Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/PaimonCatalogRangerSparkExtensionSuite.scala:54]: Solved.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/PaimonCatalogRangerSparkExtensionSuite.scala (hunk 1 above)
Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/PaimonCatalogRangerSparkExtensionSuite.scala:33]: > This is not directly config for V2 catalog implementation of Paimon (see relative configs preparation in beforeAll). Similar legacy config for "spark.sql.catalogImplementation" could be found in suites for Iceberg and Hudi.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/PaimonCatalogRangerSparkExtensionSuite.scala [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    assume(isSupportedVersion)
+    test()
+  }
+
+  override def beforeAll(): Unit = {
+    if (isSupportedVersion) {
+      spark.conf.set(s"spark.sql.catalog.$catalogV2", "org.apache.paimon.spark.SparkCatalog")
+      spark.conf.set(
+        s"spark.sql.catalog.$catalogV2.warehouse",
+        Utils.createTempDir("iceberg-hadoop").toString)
```
Comment (SteNicholas) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/PaimonCatalogRangerSparkExtensionSuite.scala:50]:
```suggestion
        Utils.createTempDir("paimon-hadoop").toString)
```

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Review by cxzl25 — APPROVED** [review]

**Comment by bowenliang123** [comment]
Thanks, merged to master (1.9.0) .

**Pull request closed without merging**